tests/functional/test_pep660.py

```diff
@@ -119,9 +129,33 @@ def test_install_pep660_basic(tmpdir: Path, script: PipTestEnvironment) -> None:
         "--no-build-isolation",
         "--editable",
         project_dir,
+        "--config-settings",
+        "x=y",
+    )
+    _assert_hook_called(project_dir, "prepare_metadata_for_build_editable")
+    _assert_hook_called_with_config_settings(project_dir, "build_editable", {"x": "y"})
+    assert (
+        result.test_env.site_packages.joinpath("project.egg-link")
+        not in result.files_created
+    ), "a .egg-link file should not have been created"
+
+
+def test_install_pep660_from_req_file(tmpdir: Path, script: PipTestEnvironment) -> None:
+    """
+    Test with backend that supports build_editable.
+    """
+    project_dir = _make_project(tmpdir, BACKEND_WITH_PEP660, with_setup_py=False)
+    reqs_file = tmpdir / "requirements.txt"
+    reqs_file.write_text(f"-e {project_dir.as_uri()} --config-setting x=y\n")
```

Comment:
You can set `--config-setting{,s}`? That was new to me. Probably set the latter for consistency?

```suggestion
    reqs_file.write_text(f"-e {project_dir.as_uri()} --config-settings x=y\n")
```

Reply:
Yes, both work, and actually the singular option is better, as each instance of the option sets one setting, and the option can be repeated. That is what `build` does and I think there is an open issue to document the singular option in `pip --help`.

Reply:
That is good to know, thanks!